core/src/main/java/io/lionweb/lioncore/java/serialization/FlatBuffersSerialization.java

```diff
@@ -0,0 +1,351 @@
+package io.lionweb.lioncore.java.serialization;
+
+import com.google.flatbuffers.FlatBufferBuilder;
+import io.lionweb.lioncore.java.language.LionCoreBuiltins;
+import io.lionweb.lioncore.java.model.ClassifierInstance;
+import io.lionweb.lioncore.java.model.impl.ProxyNode;
+import io.lionweb.lioncore.java.self.LionCore;
+import io.lionweb.lioncore.java.serialization.data.*;
+import io.lionweb.lioncore.java.serialization.data.MetaPointer;
+import io.lionweb.lioncore.java.serialization.flatbuffers.*;
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class FlatBuffersSerialization extends AbstractSerialization {
+
+  private static final String NULL_CONSTANT = "NULL";
+
+  /** This has specific support for LionCore or LionCoreBuiltins. */
+  public static FlatBuffersSerialization getStandardSerialization() {
+    FlatBuffersSerialization serialization = new FlatBuffersSerialization();
+    serialization.classifierResolver.registerLanguage(LionCore.getInstance());
+    serialization.instantiator.registerLionCoreCustomDeserializers();
+    serialization.primitiveValuesSerialization
+        .registerLionBuiltinsPrimitiveSerializersAndDeserializers();
+    serialization.instanceResolver.addAll(LionCore.getInstance().thisAndAllDescendants());
+    serialization.instanceResolver.addAll(LionCoreBuiltins.getInstance().thisAndAllDescendants());
+    return serialization;
+  }
+
+  /** This has no specific support for LionCore or LionCoreBuiltins. */
+  public static FlatBuffersSerialization getBasicSerialization() {
+    FlatBuffersSerialization serialization = new FlatBuffersSerialization();
+    return serialization;
+  }
+
+  public List<io.lionweb.lioncore.java.model.Node> deserializeToNodes(byte[] bytes)
+      throws IOException {
+    ByteBuffer bb = ByteBuffer.wrap(bytes);
+    return deserializeToNodes(FBChunk.getRootAsFBChunk(bb));
+  }
+
+  public List<io.lionweb.lioncore.java.model.Node> deserializeToNodes(FBChunk chunk) {
+    return deserializeToClassifierInstances(chunk).stream()
+        .filter(ci -> ci instanceof io.lionweb.lioncore.java.model.Node)
+        .map(ci -> (io.lionweb.lioncore.java.model.Node) ci)
+        .collect(Collectors.toList());
+  }
+
+  public List<ClassifierInstance<?>> deserializeToClassifierInstances(FBChunk chunk) {
+    SerializedChunk serializationBlock = deserializeSerializationChunk(chunk);
+    validateSerializationBlock(serializationBlock);
+    return deserializeSerializationBlock(serializationBlock);
+  }
+
+  private class DeserializationHelper {
+
+    private IdentityHashMap<FBMetaPointer, MetaPointer> metaPointersCache = new IdentityHashMap<>();
+
+    public MetaPointer deserialize(FBMetaPointer classifier) {
+      if (classifier == null) {
+        throw new IllegalStateException("Classifier should not be null");
+      }
+      if (!metaPointersCache.containsKey(classifier)) {
```

Comment:
Maybe use `computeIfAbsent()`? https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-

Reply:
Good point, done in latest commit

Reply:
(Sorry, I did not push it to the right branch, now I did)